tests/test.ts

```diff
@@ -429,3 +429,139 @@ describe('Test clasp logout function', () => {
     expect(dotExists).to.equal(false);
   });
 });
+
+describe('Test clasp help function (in general)', () => {
+  it('should show help correctly', () => {
+    const result = spawnSync(
+      CLASP, ['help'], { encoding: 'utf8' },
+    );
+    expect(result.stdout).to.contain('Usage: clasp clasp <command> [options]');
+    expect(result.status).to.equal(1);
+  });
+  it('should show help correctly', () => {
+    const result = spawnSync(
+      CLASP, ['--help'], { encoding: 'utf8' },
+    );
+    expect(result.stdout).to.contain('Usage: clasp clasp <command> [options]');
+    expect(result.status).to.equal(0);
+  });
+  it('should show help correctly', () => {
+    const result = spawnSync(
+      CLASP, ['-h'], { encoding: 'utf8' },
+    );
+    expect(result.stdout).to.contain('Usage: clasp clasp <command> [options]');
+    expect(result.status).to.equal(0);
+  });
+});
+describe('Test clasp version function', () => {
+  it('should show version correctly', () => {
+    const result = spawnSync(
+      CLASP, ['--version'], { encoding: 'utf8' },
+    );
+    expect(result.stdout).to.contain(require('./../package.json').version);
+    expect(result.status).to.equal(0);
+  });
+  it('should show help correctly', () => {
+    const result = spawnSync(
+      CLASP, ['-v'], { encoding: 'utf8' },
+    );
+    expect(result.stdout).to.contain(require('./../package.json').version);
+    expect(result.status).to.equal(0);
+  });
+});
+describe('Test clasp unknown function', () => {
+  it('should show version correctly', () => {
+    const result = spawnSync(
+      CLASP, ['unknown'], { encoding: 'utf8' },
+    );
+    expect(result.stderr).to.contain('🤔  Unknown command "unknown"');
+    expect(result.status).to.equal(1);
+  });
+});
+
+describe('Test all functions while logged out', () => {
+  it('should list error correctly', () => {
+    const result = spawnSync(
+      CLASP, ['list'], { encoding: 'utf8' },
+    );
+    expect(result.status).to.equal(1);
+    expect(result.stderr).to.contain('Could not read API credentials. Error:');
+  });
+  it('should clone error correctly', () => {
+    const result = spawnSync(
+      CLASP, ['clone'], { encoding: 'utf8' },
+    );
+    expect(result.status).to.equal(1);
+    expect(result.stderr).to.contain('Could not read API credentials. Error:');
+  });
+  it('should pull error correctly', () => {
+    const result = spawnSync(
+      CLASP, ['pull'], { encoding: 'utf8' },
+    );
+    expect(result.status).to.equal(1);
+    expect(result.stderr).to.contain('No .clasp.json settings found. `create` or `clone` a project first.');
```

Comment:
Some commands are inconsistent of the order they check for `.clasp.json` vs `.clasprs.json`. This command looks for `.clasp.json` first:

https://github.com/google/clasp/blob/87b8de5ed6b31dfbbf145ba121b41ec18404d45d/src/commands.ts#L34

Before looking for credentials in the `fetchProject(...)` method that it calls here:

https://github.com/google/clasp/blob/87b8de5ed6b31dfbbf145ba121b41ec18404d45d/src/files.ts#L183

Should probably make the order of checking for things more consistent.

Reply:
Let's create a separate issue and write down the order in which we should do things in a dev README comment like "The structure of a `clasp` command".

Reply:
You can put a comment to the inconsistent ordering here too.